File: graphs/dfs.cpp
#include<iostream>
#include<vector>
using namespace std;
void printGraph(vector<vector<int> >& g){
    for(int i=0;i<g.size();i++){
        for(int j=0;j<g.size();j++){
            cout<<g[i][j]<<"\t";
        }

        cout<<endl;
    }
}
int main(){
    int n,temp;
    cout<<"enter the number of vertices in the graph: ";
    cin>>n;
    vector<vector<int> > graph;
    cout<<"enter the adjacency matrix \n";
    for(int i=0;i<n;i++){
        cout<<"enter the data for the vertex "<<i<<endl;
        //graph.push_back(new vector<int>);
        for(int j=0;j<n;j++){
            cin>>temp;
            graph[i].push_back(temp);
        }
    }
    printGraph(graph);


    return 0;
}

File: graphs/dfs.py

def depth_first_search(m,s):
    vis = [False]*len(m)
    stack = [s-1]
    while stack:
        node = stack.pop()
        print(node)
        vis[node]=True
        for i in range(len(m)):
            if m[node][i]==1 and not vis[i]:
                stack.append(i)
def main():
    print('enter the no of vertices of the graph')
    n = int(input())
    print('enter the adjacency matrix data')
    matrix=[]
    for i in range(len(n)):
        print('enter the data into the row',i+1)
        matrix.append([])
        for j in range(len(n)):
            matrix[-1].append(int(input()))
    start = int(input('enter the starting node to explore'))
    depth_first_search(matrix,start)



if __name__ == "__main__":
    main()